jni/src/faiss_index_service.cpp

```diff
@@ -57,6 +57,104 @@ void SetExtraParameters(knn_jni::JNIUtilInterface * jniUtil, JNIEnv *env,
 
 IndexService::IndexService(std::unique_ptr<FaissMethods> faissMethods) : faissMethods(std::move(faissMethods)) {}
 
+jlong IndexService::initIndex(
+        knn_jni::JNIUtilInterface * jniUtil,
+        JNIEnv * env,
+        faiss::MetricType metric,
+        std::string indexDescription,
+        int dim,
+        int numVectors,
+        int threadCount,
+        std::unordered_map<std::string, jobject> parameters
+    ) {
+    // Removed the check for number of vectors
+    // Don't use unique_ptr here since we want to access the index in the future.
```

Comment:
When using std::unique_ptr, the object it manages is automatically destroyed when the unique_ptr goes out of scope or is reset. This automatic management ensures that resources are freed, preventing memory leaks.

It has lots of other benefits over naked pointer.

However to address the concern of getting aceess later , we can transfer the ownership . Let me give you refactored code on how to use unique pointer to get automatic memory free and don't loose transfer.